### PR TITLE
unify formatting for suggesting Gradle documentation

### LINKED ISSUE #24567: Unify formatting for suggesting Gradle documentation links
We use different phrasing to present link documentation in error messages. It's preferable to use a consistent structure to make it easier to read the message:

Proposed format: 
- For more information about `<topic>`, please refer to `<link>` in the Gradle documentation.

### PR DESCRIPTION
**Pull request by reinsch82**
- Fixes #24567

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test RFN

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

File: subprojects/logging/src/main/java/org/gradle/api/internal/DocumentationRegistry.java [hunk 1]
```diff
@@ -25,33 +25,63 @@
...
+        return BASE_URL + "/samples";
     }
 
     public String getSampleFor(String id) {
-        return String.format("https://docs.gradle.org/%s/samples/sample_%s.html", gradleVersion.getVersion(), id);
+        return String.format(getSampleIndex() + "/sample_%s.html", id);
+    }
+
+    public String getSampleForMessage(String id) {
+        return String.format("Learn more about Gradle by exploring Samples at %s", getSampleFor(id));
```
Comment (tresat) [inline subprojects/logging/src/main/java/org/gradle/api/internal/DocumentationRegistry.java:69]: The here is slightly imperative, but in the same way that “Run with --stacktrace”, or “Consult the upgrade guide” etc. is.  I don’t think anyone would find it too condescending.

You could soften it a little by saying “To learn more about” if you wanted.  You should also definitely say “exploring the Samples” or “exploring our Samples”.

**Review by tresat — COMMENTED** [review]
Didn't review the whole thing, but left a couple quick suggestions.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/ParallelStaleOutputIntegrationTest.groovy [hunk 2]
```diff
@@ -87,11 +86,15 @@ class ParallelStaleOutputIntegrationTest extends AbstractIntegrationSpec {
...
+            result.assertHasErrorOutput("Resolution of the configuration :a:myconf was attempted from a context different than the project context. " + +getDocLinkMessage());
         } else if (result.error.contains("Could not create task ':b:bar'.")) {
-            result.assertHasErrorOutput("Resolution of the configuration :b:myconf was attempted from a context different than the project context. See: https://docs.gradle.org/${GradleVersion.current().version}/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors for more information.")
+            result.assertHasErrorOutput("Resolution of the configuration :b:myconf was attempted from a context different than the project context. " + getDocLinkMessage())
         } else {
             throw new AssertionError("Unexpected task failure in test, see error output.")
         }
     }
+
+    private getDocLinkMessage() {
```
Comment (donat) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/ParallelStaleOutputIntegrationTest.groovy:97]:
```suggestion
    private def getDocLinkMessage() {
```

File: subprojects/launcher/src/main/java/org/gradle/launcher/cli/DebugLoggerWarningAction.java [hunk 3]
```diff
@@ -29,22 +29,14 @@
...
-        sb.append('\n');
-        sb.append("   ").append(new DocumentationRegistry().getDocumentationFor("logging", "sec:debug_security")).append('\n');
-        sb.append("#############################################################################\n");
-        WARNING_MESSAGE_BODY = sb.toString();
-    }
-
+    static final String WARNING_MESSAGE_BODY = "\n" +
+        "#############################################################################\n" +
+        "   WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING\n" +
+        "\n"+
```
Comment (donat) [inline subprojects/launcher/src/main/java/org/gradle/launcher/cli/DebugLoggerWarningAction.java:35]: 🧐 
```suggestion
        "\n" +
```

**Review by donat — CHANGES_REQUESTED** [review]
It looks really good!
I've spotted a few minor things, ptal.

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle clean

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**